DirectProgramming/C++SYCL_FPGA/Tutorials/Features/hls_flow_interfaces/mmhost/part3_hosts/src/mmhost.cpp

```diff
@@ -81,7 +86,14 @@ int main(void) {
     int *array_b = sycl::aligned_alloc_shared<int>(
         kAlignment, kN, q,
         sycl::ext::intel::experimental::property::usm::buffer_location(kBL2));
-    int *array_c = sycl::aligned_alloc_shared<int>(
+
+    // Allocate USM shared memory using the utility function `alloc_annotated`
+    // (defined in "annotated_class_util.hpp"), which takes an annotated_arg
+    // type as the template parameter and returns an instance of such
+    // annotated_arg. This ensures the properties of the returned memory
+    // (e.g. buffer location, alignment) match with the annotations on the
+    // kernel arguments
+    auto array_c = fpga_tools::alloc_annotated<karg_z_t>(
```

Comment:
```suggestion
    karg_z_t array_c = fpga_tools::alloc_annotated<karg_z_t>(
```

I don't like using `auto` because it obfuscates type info.

Reply:
Updated